draft-ietf-httpbis-safe-method-w-body.xml

```diff
@@ -238,42 +238,88 @@ q=foo&amp;limit=10&amp;sort=-published
-      resource identified by the request URI. That is, QUERY requests do not
-      alter the state of the identified resource. However, while processing a
-      QUERY request, a server can be expected to allocate computing and memory
-      resources or even create additional HTTP resources through which the
-      response can be retrieved.
+      QUERY requests are safe with regard to the target resource
+      (<xref target="HTTP" sectionFormat="comma" section="9.2.1"/>) --
+      that is, the client does not request or expect any change to the
+      state of the target resource. This does not prevent the server
+      from creating additional HTTP resources through which additional
+      information can be retrieved (see sections below).
     </t>
     <t>
-      A successful response to a QUERY request is expected to provide some
-      indication as to the final disposition of the operation. For
-      instance, a successful query that yields no results can be represented
-      by a 204 (No Content, <xref target="HTTP" section="15.3.5"/>) response. If the response includes content,
-      it is expected to describe the results of the operation.
+      Furthermore, QUERY requests are idempotent
+      (<xref target="HTTP" sectionFormat="comma" section="9.2.2"/>) --
+      they can be retried or repeated when needed, for instance after
+      a connection failure.
     </t>
+    <t>
+      As per <xref target="HTTP" section="15.3"/>, a 2xx (Successful) response code signals that
+      the request was successfully received, understood, and accepted.
+    </t>
+    <t>
+      In particular, a 200 (OK) response indicates that the query
+      was successfully processed and the results of that processing are
+      enclosed as the response content.
+    </t>
+
+    <section title="Media Types and Content Negotiation" anchor="media.types.and.conneg">
+      <t>
+        The semantics of a QUERY request depends both on the request content and the associated
+        metadata, such as the Media Type (<xref target="HTTP" sectionFormat="comma" section="8.3.1"/>).
+        In general, any problem with requests where content and metadata are inconsistent &MUST; be
+        rejected with a 4xx (Client Error) response (<xref target="HTTP" sectionFormat="comma" section="15.5"/>).
+      </t>
+      <t>
+        The list below describe various cases of failures and recommends specific status codes:
+      </t>
+      <ul>
+        <li>
+          A request lacking media type information by definition is incorrect and needs to fail
+          with a 4xx status code such as 400 (Client Error).
+        </li>
+        <li>
+          If a media type is specified, but not supported by the resource, a 415 (Unsupported Media Type)
+          is appropriate. This specifically includes the case where the media type is
+          known in principle, but lacks semantics specific to QUERY (for example "text/xml"). In both cases,
```

Comment:
```suggestion
          known in principle, but lacks semantics specific to a QUERY to the target resource. In both cases,
```

I understand what you are trying to say here, but I think that "text/xml" is an entirely valid thing to use, if the resource knows what to do with it.  The reason to use 415 is that *this particular resource* cannot process *this specific media type* as a QUERY.